lib/OA/Dal/Delivery/mysqli.php

```diff
@@ -0,0 +1,217 @@
+<?php
+
+/*
++---------------------------------------------------------------------------+
+| Revive Adserver                                                           |
+| http://www.revive-adserver.com                                            |
+|                                                                           |
+| Copyright: See the COPYRIGHT.txt file.                                    |
+| License: GPLv2 or later, see the LICENSE.txt file.                        |
++---------------------------------------------------------------------------+
+*/
+
+/**
+ * The mysqli data access layer code the delivery engine.
+ *
+ * @package    OpenXDal
+ * @subpackage Delivery
+ */
+
+/**
+ * The function to open a database connection, or return the resource if already open
+ *
+ * @param string $database   The name of the database config to use
+ *                           (Must match the database section name in the conf file)
+ * @return mysqli|false    The MySQL database resource
+ *                           or false on failure
+ */
+function OA_Dal_Delivery_connect($database = 'database') {
+    // If a connection already exists, then return that
+    if ($database == 'database' && isset($GLOBALS['_MAX']['ADMIN_DB_LINK']) && is_resource($GLOBALS['_MAX']['ADMIN_DB_LINK'])) {
```

Comment:
This all (until line 49) looks generic to me and not MySQLi-specific. In my view this should be moved out to a more general class (yes, this is function-based code, not object-oriented).